llama_index/llm_predictor/base.py

```diff
@@ -131,17 +128,75 @@ def _log_template_data(
         ):
             pass
 
-    def predict(self, prompt: BasePromptTemplate, **prompt_args: Any) -> str:
+    def _run_program(
+        self,
+        output_cls: BaseModel,
+        prompt: PromptTemplate,
+        **prompt_args: Any,
+    ) -> str:
+        program: BasePydanticProgram
+        try:
+            from llama_index.program.openai_program import OpenAIPydanticProgram
+
+            program = OpenAIPydanticProgram.from_defaults(
+                output_cls=output_cls,
+                llm=self._llm,
+                prompt=prompt,
+            )
+        except ValueError:
+            from llama_index.program.llm_program import LLMTextCompletionProgram
+
+            program = LLMTextCompletionProgram.from_defaults(
+                output_parser=PydanticOutputParser(output_cls=output_cls),
+                llm=self._llm,
+                prompt=prompt,
+            )
+        chat_response = program(**prompt_args)
+        return chat_response.json()
+
+    async def _arun_program(
+        self,
+        output_cls: BaseModel,
+        prompt: BasePromptTemplate,
```

Comment:
nit: ooc why is the async prompt typed differently than the sync version

Reply:
Fixed! Thanks.